src/expressions/functions/builtInFunctions_node.ts

```diff
@@ -165,6 +165,78 @@ function areSameNode(
 	return true;
 }
 
+/**
+ * Test if a string should match the xml:lang attribute string of a node.
+ */
+function langMatchString(truth: string, test: string): boolean {
+	truth = truth.toLowerCase();
+	test = test.toLowerCase();
+	if (truth === test) {
+		return true;
+	} else {
+		return truth.length < 5 || !truth.startsWith(test)
+			? false
+			: /* Strip the least specific part and test again */
+			  langMatchString(truth.replace(/-[a-z0-9]+$/, ''), test);
+	}
+}
+
+/**
+ * Starting at a given node, test the xml:lang attribute to see if it
+   matches a test. Move up the tree until the test passes or fails.
+ */
+function langMatchNode(node: NodePointer, domFacade: DomFacade, test: string): ISequence {
+	let toTest = node;
+	while (toTest) {
+		if (domFacade.getNodeType(toTest) !== NODE_TYPES.ELEMENT_NODE) {
+			toTest = domFacade.getParentNodePointer(toTest as AttributeNodePointer);
+		} else if (domFacade.getNodeType(toTest) === NODE_TYPES.ELEMENT_NODE) {
+			const attrValue = domFacade.getAttribute(toTest as ElementNodePointer, 'xml:lang');
```

Comment:
Should this be a namespace-aware getAttributeNS?

Reply:
Weirdly, it's required to use the `xml` prefix, unlike other namespaces. https://www.w3.org/XML/1998/namespace

Reply:
Ah true, that would make this equivalent 👍